lib/auth/accountrecovery_test.go

```diff
@@ -111,3 +133,323 @@ func TestRecoveryCodeEventsEmitted(t *testing.T) {
 	require.Equal(t, events.RecoveryCodeUsedEvent, event.GetType())
 	require.Equal(t, events.RecoveryCodeUseFailureCode, event.GetCode())
 }
+
+func TestStartAccountRecovery(t *testing.T) {
+	srv := newTestTLSServer(t)
+	ctx := context.Background()
+	fakeClock := srv.Clock().(clockwork.FakeClock)
+	mockEmitter := &events.MockEmitter{}
+	srv.Auth().emitter = mockEmitter
+
+	defaultModules := modules.GetModules()
+	defer modules.SetModules(defaultModules)
+	modules.SetModules(&testWithCloudModules{})
+
+	u, err := createUserWithSecondFactorAndRecoveryCodes(srv, "otp")
+	require.NoError(t, err)
+
+	// Test with no recover type.
+	_, err = srv.Auth().StartAccountRecovery(ctx, &proto.StartAccountRecoveryRequest{
+		Username:     u.username,
+		RecoveryCode: []byte(u.recoveryCodes[0]),
+	})
+	require.Error(t, err)
+
+	cases := []struct {
+		desc         string
```

Comment:
Suggestion: I think the more usual form of this is `name string` ([example from Go wiki](https://github.com/golang/go/wiki/TableDrivenTests#parallel-testing)).

```suggestion
		name         string
```